Kernel/Syscalls/execve.cpp

```diff
@@ -306,7 +306,7 @@ static ErrorOr<LoadResult> load_elf_object(NonnullOwnPtr<Memory::AddressSpace> n
 
     auto load_writable_section = [&](auto& program_header) -> ErrorOr<void> {
         // Writable section: create a copy in memory.
-        VERIFY(program_header.alignment() == PAGE_SIZE);
+        VERIFY(program_header.alignment() % PAGE_SIZE == 0);
```

Comment:
This is not enough, the call to `try_allocate_range` has to be updated to take in the requested alignment, since the default value of `PAGE_SIZE` is no longer always accurate

Reply:
Sure enough, the goal for this PR was to resolve the kernel panic but it makes sense to pass along the program header's alignment as well. This does not fix the ensuing userspace crash though, which needs to be fixed along the lines of "what does it mean to execute an `.so`" (this should probably not be possible).

I've created a separate commit for your proposed chance, since 1) it is not kernel panic related and 2) we are now passing in a program header's alignment (i.e. user input from arbitrarily modifiable `.so` files), which is an `u32`, to an allocation method which should be separately revertible imho.

Let me know what you think!

Reply:
I agree with the separate commit, I don't agree with the commit order, ignoring the specific `.so` file crash you're fixing, the first commit essentially introduces a bug, and the second one fixes it (There may be legitimate executable that request a higher alignment, and depend on it, which will be loaded incorrectly by the `try_allocate_range` call)

Reply:
I could reorder the commits, but since there is no way `try_allocate_range` could get anything other than the default `PAGE_SIZE` before the second commit I decided to merge them.